libvast/include/vast/detail/string_literal.hpp

```diff
@@ -0,0 +1,33 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <string_view>
+
+namespace vast::detail {
+
+/// String literal wrapper for NTTPs.
```

Comment:
Googling doesn't reveal anything meaningful for me here. Can this be made less obscure? Maybe add a URL?

Reply:
It's literally what the comment says: a class making string literals usable as non-type template parameters. A similar technique exists in {fmt}, just to list one example. It's a reallx common meta-progrmaming technique.

NB: I actually wrote this last year when I first dabbled with `co_result<T>`, and this is a carbon-copy from that PR.

Reply:
I expanded the comment above this class to make this a bit clearer.